### ISSUE TITLE
对于class PretrainDataset(Dataset)的一些疑惑

### ISSUE THREAD

**Issue by chenggguang** [jingyaogong/minimind#648]
def __getitem__(self, index):
        sample = `self.samples[index]`
        tokens = self.tokenizer(str(sample['text']), add_special_tokens=False, max_length=self.max_length - 2, truncation=True).input_ids
        tokens = [self.tokenizer.bos_token_id] + tokens + [self.tokenizer.eos_token_id]
        input_ids = tokens + [self.tokenizer.pad_token_id] * (self.max_length - len(tokens))
        input_ids = torch.tensor(input_ids, dtype=torch.long)
        labels = input_ids.clone()
        labels[input_ids == self.tokenizer.pad_token_id] = -100
        return input_ids, labels

代码中， 先tokenize了文本，然后加上了起始 token, 如果是普通文本，这个应该没有问题
但是在 pretrain_hq.jsonl 中， 每一个样本的文本都有特殊token,这样操作之后的效果就会变成 

`[1, 1, x, x, x, x, x, 2, 2]`

特殊token会重复

附上一个 pretrain_hq.jsonl 文本行

> {"text": "<|im_start|>好的，接下来请你将计算结果四舍五入到小数点后两位。好的，计算结果四舍五入到小数点后两位即为9.00。<|im_end|> <|im_start|>非常感谢。现在请你统计一下这段话中“中国”这个词出现的次数。在这段文本中，“中国”这个词出现了两次。<|im_end|> <|im_start|>你好，我遇到了一个很棘手的问题，能请你帮帮我吗？当然，我会尽力帮助你解决问题。请问你遇到了什么困难？<|im_end|> <|im_start|>麦哲伦环球航行是哪一年？麦哲伦的航行开始于1519年，而完成环球航行的船只则于1522年返回西班牙。<|im_end|> <|im_start|>写一首五言古诗，以“春日”为主题。万物复苏春意浓，和风拂面桃花红。\n婉转笛声飘月殿，微笑春风抚徐公。<|im_end|> <|im_start|>帮我推荐一本好看的小说。《放学后》，这是一本关于高中生活与友情的小说，情节引人入胜，适合青少年阅读。<|im_end|> <|im_start|>嗯，这个建议还不错。那加个“花香扑鼻”如何？不错的建议！这样可以更加形象地表达出爱情中的美好和浪漫。<|im_end|> <|im_start|>那么给我一个肯定的回答是否下雨了？是的，昨天下了雨。\n详细回答可以是：\n昨天有降雨，天气也比较阴沉。<|im_end|>"}

**Comment by jingyaogong** [comment]
更新这个代码的时候， `pretrain_hq.jsonl` 也更新过了，你展示的是「新代码+旧数据」
